frontend/src/components/intake/ShasIntakeSchema.ts

```diff
@@ -0,0 +1,144 @@
+      financiallySupportedBC: YesNoUnsureSchema.label('BC Housing'),
+      financiallySupportedIndigenous: YesNoUnsureSchema.label('Indigenous Housing Provider'),
+      financiallySupportedNonProfit: YesNoUnsureSchema.label('Non-profit housing society'),
+      financiallySupportedHousingCoop: YesNoUnsureSchema.label('Housing co-operative'),
+      rentalUnits: string().when('hasRentalUnits', {
+        is: prevYesSelected,
+        then: () => string().oneOf(NumResidentialUnits).required().label('Expected rental units'),
+        otherwise: () => string().nullable()
+      }),
+      indigenousDescription: string().when('financiallySupportedIndigenous', {
+        is: prevYesSelected,
+        then: () => stringRequired.label('Indigenous housing provider'),
+        otherwise: () => string().nullable()
+      }),
+      nonProfitDescription: string().when('financiallySupportedNonProfit', {
+        is: prevYesSelected,
+        then: () => stringRequired.label('Non-profit housing society'),
+        otherwise: () => string().nullable()
+      }),
+      housingCoopDescription: string().when('financiallySupportedHousingCoop', {
+        is: prevYesSelected,
+        then: () => stringRequired.label('Housing co-operative'),
+        otherwise: () => string().nullable()
+      }),
+      singleFamilySelected: boolean(),
+      singleFamilyUnits: string().when('singleFamilySelected', {
+        is: (prev: boolean) => prev,
+        then: (schema) => schema.required().oneOf(NumResidentialUnits).label('Expected number of single-family units'),
+        otherwise: () => string().nullable()
+      }),
+      multiFamilySelected: boolean(),
+      multiFamilyUnits: string().when('multiFamilySelected', {
+        is: (prev: boolean) => prev,
+        then: (schema) => schema.required().oneOf(NumResidentialUnits).label('Expected number of multi-family units'),
+        otherwise: () => string().nullable()
+      }),
+      otherSelected: boolean(),
+      otherUnitsDescription: string().when('otherSelected', {
+        is: (prev: boolean) => prev,
+        then: (schema) => schema.required().label('Description of units'),
+        otherwise: () => string().nullable()
+      }),
+      otherUnits: string().when('otherSelected', {
+        is: (prev: boolean) => prev,
+        then: (schema) => schema.required().oneOf(NumResidentialUnits).label('Expected number of other units'),
+        otherwise: () => string().nullable()
+      })
+    }).test('housing-checkbox-test', 'At least one residential type must be selected', (value, context) => {
+      return (
+        context.originalValue.singleFamilySelected ||
+        context.originalValue.multiFamilySelected ||
+        context.originalValue.otherSelected
+      );
+    }),
+    location: object({
+      projectLocation: string().required().label('Location'),
+      streetAddress: string().when('projectLocation', {
+        is: (prevQuestion: string) => prevQuestion === ProjectLocation[0],
```

Comment:
Import the constants in this file and check directly against them. If someone were to inadvertently flip the array values around the validation would explode.